features/org.openhab.deps.runtime/feature.xml

```diff
@@ -24,7 +24,7 @@
          id="com.google.guava"
          download-size="0"
          install-size="0"
-         version="15.0.0.v201403281430"
+         version="21.0.0.v20170206-1425"
```

Comment:
Wasn’t there some semver related problem with this version bump of guava or is our current use so limited that it’s unlikely that anything will break here.

Reply:
I'm not aware of these issues myself. We've also added the packages to the [forbidden packages](https://github.com/openhab/openhab2-addons/blob/1a6a1be6cae8d9b1da31fef693a82f80b70bb65a/tools/checkstyle.properties#L4) in SAT and done several PRs to reduce the usage.

It seems Guava 18.0.0 is also still available:

```
openhab> bundle:list -t 0 -s | grep guava
 22 │ Active   │  80 │ 18.0.0                  │ com.google.guava
 23 │ Active   │  80 │ 21.0.0                  │ com.google.guava
122 │ Active   │  36 │ 2.22.2                  │ org.glassfish.jersey.bundles.repackaged.jersey-guava
```

Without also adding Guava 21.0 it's not possible to upgrade Xtext to 2.17 (required for Java 11 support, https://github.com/openhab/openhab-core/pull/650).

They've now pinned the version to 21.0 (see "Guava 21.0 Version Pinning" in the [release notes](https://www.eclipse.org/Xtext/releasenotes.html#/releasenotes/2019/03/05/version-2-17-0)).

If we run into issues we can of course also provide an even older Guava version. But Guava is quite big so I don't think it's desirable to provide unnecessary versions.

Reply:
Some time ago (Eclipse SmartHome ones):

* Xtext needs min. 15: https://github.com/eclipse/smarthome/blob/master/features/karaf/esh-tp/src/main/feature/feature.xml#L174
* Swagger JAX RS Provider needs 18: https://github.com/eclipse/smarthome/blob/master/features/karaf/esh-tp/src/main/feature/feature.xml#L233
* ESH itself works with >= 14

@wborn Why do you need 18 and 21?
Which bundle (Swagger JAX RS Provider?) uses a max. version for Guava?

I already worked on the removal of Guava in ESH a time before, see: https://github.com/eclipse/smarthome/pull/6449 and the linked one https://github.com/eclipse/smarthome/issues/6460

Reply:
It's indeed a Swagger requirement:

```
io.swagger.jaxrs_1.5.8 [216] requires:
--------------------------------------
...
osgi.wiring.package; (&(osgi.wiring.package=com.google.common.base)(&(version>=18.0.0)(!(version>=19.0.0)))) resolved by:
   osgi.wiring.package; com.google.common.base 18.0.0 from com.google.guava_18.0.0 [22]
osgi.wiring.package; (&(osgi.wiring.package=com.google.common.collect)(&(version>=18.0.0)(!(version>=19.0.0)))) resolved by:
   osgi.wiring.package; com.google.common.collect 18.0.0 from com.google.guava_18.0.0 [22]
...
```

Maybe there's an updated bundle so we can just provide 21.

Reply:
The main reason why the swagger POMs define an older Guava version is because Swagger 1.5.x is [compatible with Java 7](https://github.com/swagger-api/swagger-core/tree/1.5#prerequisites). Guava 21.0 is the first version that [requires Java 8](https://github.com/google/guava/wiki/Release21#java-8).